src/node_util.cc

```diff
@@ -43,6 +44,22 @@ using v8::Value;
   VALUE_METHOD_MAP(V)
 #undef V
 
+static void GetPromiseDetails(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsPromise());
+
+  auto isolate = args.GetIsolate();
+
+  Local<Promise> promise = args[0].As<Promise>();
+  Local<Array> ret = Array::New(isolate, 2);
```

Comment:
might it be better to use Object and name the properties? might be easier if we want to extend with additional metadata, and would allow the JavaScript to be de-coupled from the order of items in the array.